packages/Caching/ValueObject/Storage/FileCacheStorage.php

```diff
@@ -24,59 +23,60 @@ public function __construct(
     }
 
     /**
+     * @param string $key
+     * @param string $variableKey
      * @return mixed|null
      */
     public function load(string $key, string $variableKey)
     {
-        $cacheFilePaths = $this->getCacheFilePaths($key);
-
-        $filePath = $cacheFilePaths->getFilePath();
-        if (! is_file($filePath)) {
-            return null;
-        }
-
-        $cacheItem = require $filePath;
-        if (! $cacheItem instanceof CacheItem) {
-            return null;
-        }
-
-        if (! $cacheItem->isVariableKeyValid($variableKey)) {
-            return null;
-        }
-
-        return $cacheItem->getData();
+        return (function (string $key, string $variableKey) {
+            $cacheFilePaths = $this->getCacheFilePaths($key);
+
+            $filePath = $cacheFilePaths->getFilePath();
+            if (!\is_file($filePath)) {
+                return null;
+            }
+            $cacheItem = (require $filePath);
+            if (!$cacheItem instanceof CacheItem) {
+                return null;
+            }
+            if (!$cacheItem->isVariableKeyValid($variableKey)) {
+                return null;
+            }
+            return $cacheItem->getData();
+        })($key, $variableKey);
```

Comment:
This is very confusing to read. Not sure if args are on left or right.
Could you extract function to assign line and invoke it on 2nd?

Reply:
its the way how phpstan does this things.

all these nitpicks are great, but it goes against intially mentioned goal:

> Current design is based on cache in PHPStan, and I would like to keep it as close the original to make the maintenance easier.

after we apply all 'rector rules' the diff between the phpstan impl and ours here gets bigger and bigger.

Reply:
Ok, you're right. Let's give it a go

Reply:
fwiw, I opened a upstream PR to simplify this on phpstan-src

https://github.com/phpstan/phpstan-src/pull/593

Reply:
Cool! :+1: 

This is too much magic for me. I understand only objects and methods :D